Document new create_mirror[_view] overloads taking Kokkos::view_alloc() return type (#105)
kokkos/kokkos-core-wiki: docs/source/API/core/view/create_mirror.md

ImplMirrorType create_mirror_view(decltype(Kokkos::ViewAllocateWithoutInitializing()),
                                  Space const& space, ViewType const&);

template <class ViewType, class... ViewCtorArgs>

> auto create_mirror_view(Impl::ViewCtorProp<ViewCtorArgs...> const& arg_prop,

Reviewer: Can we try something like:

```c++
template <class ViewType, class... ViewCtorArgs>
 auto create_mirror_view(ALLOC_PROP const& arg_prop, ViewType const& v);
```

or 

```c++
template <class ViewType, class... ViewCtorArgs>
 auto create_mirror_view(/*see below*/ const& arg_prop, ViewType const& v);
```

or

```c++
template <class ViewType, class... ViewCtorArgs>
 auto create_mirror_view(/* is-alloc-prop */ const& arg_prop, ViewType const& v);
```

Reviewer: I think each of those is better than either `Impl::ViewCtorProp`, or `decltype(ViewAllocationWithoutInitializing())`

You then in the description define what `ALLOC_PROP` or `/*is-alloc-prop*/` means.

Reviewer: I think right now it seems we have a preference for the last one? Maybe as `/*is-view-alloc-prop*/` to make it even more clear? And then spell out what all the things are which can go there?

Reviewer: Note I do not want to use this in places of the explicit "takes an execution space instance as first argument" overloads.

Reviewer 2: +1 for option 3

Author: Here you go!

  * `Space`: a class meeting the requirements of [`ExecutionSpaceConcept`](ExecutionSpaceConcept) or [`MemorySpaceConcept`](MemorySpaceConcept)
  * `ImplMirrorType`: an implementation defined specialization of `Kokkos::View`.

* ```c++
  template <class ViewType, class... ViewCtorArgs>
  auto create_mirror_view(Impl::ViewCtorProp<ViewCtorArgs...> const& arg_prop,
                          ViewType const& src);
  ```

>   If the [`View`](view) constructor arguments `arg_prop` include a memory space and the memory space doesn't match the memory space of `src`, creates a new [`View`](view) in the specified memory_space.

Reviewer 3: ```suggestion
  If the [`View`](view) constructor arguments `arg_prop` include a memory space and the memory space doesn't match the memory space of `src`, a new [`View`](view) in the specified memory_space is created.
```

>   If the `arg_prop` don't include a memory space and the memory space of `src` is not host-accessible, creates a new host-accessible [`View`](view).
>   Otherwise, `src` is returned.

Reviewer 3: ```suggestion
  If the `arg_prop` don't include a memory space and the memory space of `src` is not host-accessible, a new host-accessible [`View`](view) is created.
  Otherwise, `src` is returned.
```

  * `src`: a `Kokkos::View`.
  * `Space`: a class meeting the requirements of [`ExecutionSpaceConcept`](ExecutionSpaceConcept) or [`MemorySpaceConcept`](MemorySpaceConcept)
  * `ImplMirrorType`: an implementation defined specialization of `Kokkos::View`.

* ```c++
  template <class ViewType, class... ViewCtorArgs>
  ImplMirrorType create_mirror_view_and_copy(Impl::ViewCtorProp<ViewCtorArgs...> const& arg_prop,
                                             ViewType const& src);
  ```

>   If the  memory space included in the [`View`](view) constructor arguments `arg_prop` matches the memory space of `src`, creates a new [`View`](view) in the specified memory space using `arg_prop` and the same layout andf padding as `src`. Additionally, a `deep_copy` from `src` to the new view is executed (using the execution space contained in `arg_prop` if provided).

Reviewer 3: ```suggestion
  If the  memory space included in the [`View`](view) constructor arguments `arg_prop` matches the memory space of `src`, a new [`View`](view) in the specified memory space is created using `arg_prop` and the same layout andf padding as `src`. Additionally, a `deep_copy` from `src` to the new view is executed (using the execution space contained in `arg_prop` if provided).
```
